src/pubsub/ua_pubsub_ns0.c

```diff
@@ -1799,6 +1902,32 @@ addDataSetWriterRepresentation(UA_Server *server, UA_DataSetWriter *dataSetWrite
                             UA_NODEID_NUMERIC(0, UA_NS0ID_HASPROPERTY),
                             dataSetWriter->identifier);
 
+    UA_NodeId statusIdNode = 
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "Status"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                            dataSetWriter->identifier);
+    
+    if(UA_NodeId_isNull(&statusIdNode)) {
+        return UA_STATUSCODE_BADNOTFOUND;
+    }
+
+    UA_NodeId stateIdNode = 
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "State"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                            statusIdNode);
+
+    if(UA_NodeId_isNull(&keyFrameNode)) {
+
+        // TODO Check why keyFrameNode is null
```

Comment:
The issue was present before. I included the check to test if the value is NULL. This could be a bug to be resolved later.
As discussed I would update the comment to make it a general issue